## Overview {: .expand}

### About

GCTB is a software tool that comprises Bayesian mixed linear models for complex trait analyses using genome-wide SNPs. It was developed to simultaneously estimate the joint effects of all SNPs and the genetic architecture parameters for a complex trait, including SNP-based heritability, polygenicity and the joint distribution of effect sizes and minor allele frequencies. 

### Credits 

[Jian Zeng](http://researchers.uq.edu.au/researcher/14033) developed
the software tool with supports from [Jian Yang](https://scholar.google.com.au/citations?user=aLuqQs8AAAAJ&hl=en) and [Futao Zhang](http://researchers.uq.edu.au/researcher/12709). [Luke Lloyd-Jones](https://scholar.google.com.au/citations?user=WQ0UXOcAAAAJ&hl=en&oi=ao) contributed the BayesR module with supports from Jian Zeng and [Mike Goddard](https://en.wikipedia.org/wiki/Michael_Goddard).

### Questions and Help Requests 

If you have any bug reports or questions, please send an email to Jian Zeng (<j.zeng@uq.edu.au>) or Jian Yang (<jian.yang@uq.edu.au>).

### Citations 

Zeng, J. et al. Widespread signatures of negative selection in the genetic architecture of human complex traits 
[*bioRxiv*](https://doi.org/10.1101/145755) (2017)

**BayesC$\pi$**  
Habier, David, et al. Extension of the Bayesian alphabet for genomic selection. [*BMC Bioinformatics* 12.1](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-12-186) (2011): 186.

**BayesR**  
Moser, Gerhard, et al. Simultaneous discovery, estimation and prediction analysis of complex traits using a Bayesian mixture model. [*PLoS Genetics* 11.4](http://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1004969) (2015): e1004969.
