## Overview {: .expand}

### About

GCTB is a software tool that comprises a family of Bayesian linear mixed models for complex trait analyses using genome-wide SNPs. It was developed to simultaneously estimate the joint effects of all SNPs and the genetic architecture parameters for a complex trait, including SNP-based heritability, polygenicity and the joint distribution of effect sizes and minor allele frequencies. 

### Credits 

[Jian Zeng](http://researchers.uq.edu.au/researcher/14033) developed
the software with supports from [Jian Yang](https://scholar.google.com.au/citations?user=aLuqQs8AAAAJ&hl=en), [Futao Zhang](http://researchers.uq.edu.au/researcher/12709) and [Zhili Zheng](http://researchers.uq.edu.au/researcher/19885). Part of the code are adopted from [*GCTA*](http://cnsgenomics.com/software/gcta) and [*GenSel*](https://static-content.springer.com/esm/art%3A10.1186%2F1471-2105-12-186/MediaObjects/12859_2010_4655_MOESM1_ESM.PDF). [Luke Lloyd-Jones](https://scholar.google.com.au/citations?user=WQ0UXOcAAAAJ&hl=en&oi=ao) contributed to the BayesR module with supports from [Jian Zeng](http://researchers.uq.edu.au/researcher/14033) and [Mike Goddard](https://en.wikipedia.org/wiki/Michael_Goddard).

### Questions and Help Requests 

If you have any bug reports or questions, please send an email to Jian Zeng (<j.zeng@uq.edu.au>) or Jian Yang (<jian.yang@uq.edu.au>).

### Citation 

Zeng et al. (2018) Signatures of negative selection in the genetic architecture of human complex traits. 
[*Nature Genetics*, doi: 10.1038/s41588-018-0101-4.](https://www.nature.com/articles/s41588-018-0101-4)
