<html xmlns="http://www.w3.org/1999/xhtml"><head>
    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
    <title>GCTB</title>
    <link type="text/css" rel="stylesheet" href="./static/css/jquery.tocify.css">
    <link type="text/css" rel="stylesheet" href="./static/css/bootstrap.min.css">
    <link type="text/css" rel="stylesheet" href="./static/css/solarized-light.css">
    <link type="text/css" rel="stylesheet" href="./static/css/main.css">
    <script src="./static/js/jquery-1.12.4.min.js"></script>
    <script src="./static/js/jquery-ui.min.js"></script>
    <script src="./static/js/jquery.tocify.js"></script>
    <script src="./static/js/hljs.js"></script>
    <script type="text/javascript" async="" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.4/latest.js?config=TeX-MML-AM_CHTML">
    </script>
</head>

<body>
    <div class="page-header" id="PAGE-header" style="height:auto;padding:0">
        <div class="container-fluid">
            <p class="TOP_head">&nbsp;</p>
            <div class="span5">
                <h1>&nbsp;GCTB</h1>
                <p class="TOP_head">A tool for Genome-wide Complex Trait Bayesian analysis</p>
            </div>
            <div class="span7">
                <ul class="nav nav-pills" style="margin:0">
                    <li role="presentation"><a href="http://cnsgenomics.com/software/gcta/">GCTA</a></li>
                    <li role="presentation"><a href="http://cnsgenomics.com/software/smr/">SMR</a></li>
                    <li role="presentation"><a href="http://cnsgenomics.com/software/gsmr/">GSMR</a></li>
                    <li role="presentation"><a href="http://cnsgenomics.com/software/osca/">OSCA</a></li>
                    <li role="presentation" class="active"><a href="http://cnsgenomics.com/software/gctb/">GCTB</a></li>
                    <li role="presentation"><a href="http://cnsgenomics.com/software.html">Program in PCTG</a></li>
                    <li role="presentation"><a href="http://gcta.freeforums.net/">CTG forum</a></li>
                </ul>
            </div>
        </div>
    </div>
    <div class="container-fluid">
        <div class="row-fluid">
            <div class="span3">
                <div id="toc"></div>
                <!--[if lt IE 9]>
				<div style=" margin-top: 230px; position: fixed; width: 230px; background-color: yellow; z-index: 10000;">
				<p>Due to our shortage of testing, the document can't view correctly under Internet Explorer <= 8.0. You may need update your browser, or downlad offline <a href="./static/gcta_doc_latest.pdf">PDF document</a>, <a href="./static/gcta_latest.zip">executable Files</a></p>
				</div>
				<![endif]-->
            </div>
            <div id="show" class="span9">
                <p> Loading...</p>
                <p> If it not work properly, you may need update your Internet browser and enable javascript</p>
                <p> or downlad offline <a href="./download/GCTB.pdff">PDF document</a>, <a href="./download/gctb_1.0_Linux.zip">executable Files</a></p>
            </div>
            <div id="content" class="span9" style="display: none;"><body><h2 id="overviewexpand" class="expand">Overview </h2>
<hr><h3 id="about">About</h3>
<p>GCTB is a software tool that comprises a family of Bayesian linear mixed models for complex trait analyses using genome-wide SNPs. It was developed to simultaneously estimate the joint effects of all SNPs and the genetic architecture parameters for a complex trait, including SNP-based heritability, polygenicity and the joint distribution of effect sizes and minor allele frequencies. </p>
<hr><h3 id="credits">Credits</h3>
<p><a href="http://researchers.uq.edu.au/researcher/14033">Jian Zeng</a> developed
the software with supports from <a href="https://scholar.google.com.au/citations?user=aLuqQs8AAAAJ&amp;hl=en">Jian Yang</a>, <a href="http://researchers.uq.edu.au/researcher/12709">Futao Zhang</a> and <a href="http://researchers.uq.edu.au/researcher/19885">Zhili Zheng</a>. Part of the code are adopted from <a href="http://cnsgenomics.com/software/gcta"><em>GCTA</em></a> and <a href="https://static-content.springer.com/esm/art%3A10.1186%2F1471-2105-12-186/MediaObjects/12859_2010_4655_MOESM1_ESM.PDF"><em>GenSel</em></a>. <a href="https://scholar.google.com.au/citations?user=WQ0UXOcAAAAJ&amp;hl=en&amp;oi=ao">Luke Lloyd-Jones</a> contributed to the BayesR module with supports from <a href="http://researchers.uq.edu.au/researcher/14033">Jian Zeng</a> and <a href="https://en.wikipedia.org/wiki/Michael_Goddard">Mike Goddard</a>.</p>
<hr><h3 id="questionsandhelprequests">Questions and Help Requests</h3>
<p>If you have any bug reports or questions, please send an email to Jian Zeng (<a href="mailto:j.zeng@uq.edu.au">j.zeng@uq.edu.au</a>) or Jian Yang (<a href="mailto:jian.yang@uq.edu.au">jian.yang@uq.edu.au</a>).</p>
<hr><h3 id="citation">Citation</h3>
<p>Zeng et al. (2018) Signatures of negative selection in the genetic architecture of human complex traits. 
<em>Nature Genetics</em>, doi: 10.1038/s41588-018-0101-4.</p>
<h2 id="downloadexpand" class="expand">Download </h2>
<hr><h3 id="executablefiles">Executable Files</h3>
<p><a href="download/gctb_1.0_Linux.zip">gctb_1.0_Linux.zip</a></p>
<p><a href="download/gctb_1.0_Mac.zip">gctb_1.0_Mac.zip</a></p>
<hr><h3 id="sourcecode">Source code</h3>
<p><a href="download/gctb_1.0_scr.zip">Standard version</a></p>
<p><a href="download/gctb_1.0_mpi_scr.zip">MPI version</a></p>
<p>The MPI version implements a distributed computing strategy that scales the analysis to very large sample sizes. A significant improvement in computing time is expected for a sample size &gt; 10,000. The MPI version needs to be compiled on userâ€™s machine. See <a href="download/README.html">README.html</a> in the tarball for instructions of compilation and usage. A testing dataset is also included in each tarball.</p>
<hr><h3 id="updatelog">Update log</h3>
<p><strong>1.</strong>  1 Dec, 2017: first release.</p>
<h2 id="basicoptions">Basic options</h2>
<hr><h3 id="inputandoutput">Input and output</h3>
<p><strong><span class="kwd">--bfile</span></strong> test</p>
<p>Input PLINK binary PED files, e.g. test.fam, test.bim and test.bed (see PLINK user manual for details).</p>
<p><strong><span class="kwd">--pheno</span></strong> test.phen</p>
<p>Input phenotype data from a plain text file, e.g. test.phen.</p>
<p><strong><span class="kwd">--out</span></strong> test</p>
<p>Specify output root filename.</p>
<hr><h3 id="datamanagement">Data management</h3>
<p><strong><span class="kwd">--keep</span></strong> test.indi.list</p>
<p>Specify a list of individuals to be included in the analysis.</p>
<p><strong><span class="kwd">--chr</span></strong> 1</p>
<p>Include SNPs on a specific chromosome in the analysis, e.g. chromosome 1.</p>
<p><strong><span class="kwd">--extract</span></strong> test.snplist</p>
<p>Specify a list of SNPs to be included in the analysis.</p>
<p><strong><span class="kwd">--exclude</span></strong> test.snplist</p>
<p>Specify a list of SNPs to be excluded from the analysis.</p>
<p><strong><span class="kwd">--mpheno</span></strong> 2</p>
<p>If the phenotype file contains more than one trait, by default, GCTB takes the first trait for analysis (the third column of the file) unless this option is specified. For example, <strong><span class="kwd">--mpheno</span></strong> 2 tells GCTB to take the second trait for analysis (the fourth column of the file).</p>
<p><strong><span class="kwd">--covar</span></strong> test.qcovar</p>
<p>Input quantitative covariates from a plain text file, e.g. test.qcovar. Each quantitative covariate is recognized as a continuous variable.</p>
<hr><h3 id="mcmcsettings">MCMC settings</h3>
<p><strong><span class="kwd">--seed</span></strong> 123</p>
<p>Specify the seed for random number generation, e.g. 123. Note that giving the same seed value would result in exactly the same results between two runs.</p>
<p><strong><span class="kwd">--chain-length</span></strong> 21000</p>
<p>Specify the total number of iterations in MCMC, e.g. 21000 (default).</p>
<p><strong><span class="kwd">--burn-in</span></strong> 1000</p>
<p>Specify the number of iterations to be discarded, e.g. 1000 (default).</p>
<p><strong><span class="kwd">--out-freq</span></strong> 100</p>
<p>Display the intermediate results for every 100 iterations (default). </p>
<p><strong><span class="kwd">--thin</span></strong> 10</p>
<p>Output the sampled values for SNP effects and genetic architecture parameters for every 10 iterations (default). Only non-zero sampled values of SNP effects are written into a binary file.</p>
<p><strong><span class="kwd">--no-mcmc-bin</span></strong></p>
<p>Suppress the output of MCMC samples of SNP effects.</p>
<h2 id="bayesianalphabet">Bayesian alphabet</h2>
<p><strong><span class="kwd">--bayes</span></strong> S</p>
<p>Specify the Bayesian alphabet for the analysis, e.g. `S`. Different alphabet launch different models, which differ in the prior specification for the SNP effects. The available alphabet include</p>
<ul>
<li><p>B: Each SNP effect is assumed to have an i.i.d. mixture prior of a t-distribution `t(0, \tau^2, \nu)` with a probability `\pi` and a point mass at zero with a probability `1-\pi`.</p></li>
<li><p>C: Each SNP effect is assumed to have an i.i.d. mixture prior of a normal distribution `N(0, \sigma^2)` with a probability `\pi` and a point mass at zero with a probability `1-\pi`. </p></li>
<li><p>S: Similar to C but the variance of SNP effects is related to minor allele frequency (`p`) through a parameter `S`, i.e. `\sigma_j^2 = [2p_j(1-p_j)]^S \sigma^2`.</p></li>
<li><p>N: nested BayesC. SNPs within a 0.2 Mb non-overlapping genomic region are collectively considered as a window (specify the distance by <strong><span class="kwd">--wind</span></strong> 0.2). This nested approach speeds up the analysis by skipping over windows with zero effect.</p></li>
<li><p>NS: nested BayesS.</p></li>
<li><p>R: BayesR. Each SNP effect is assumed to have an i.i.d. mixture prior of multiple normal distributions `N(0, \gamma_k \sigma_k^2)` with a probability `\pi_k` and a point mass at zero with a probability `1-\sum_k \pi_k`, where `\gamma_k` is a given constant.</p></li>
</ul>
<p><strong><span class="kwd">--fix-pi</span></strong></p>
<p>An option to fix `\pi` to a constant (the value is specified by the option <span class="kwd">--pi</span> below). The default setting is to treat Ï€ as random and estimate it from the data.</p>
<p><strong><span class="kwd">--pi</span></strong> 0.05</p>
<p>A starting value for the sampling of Ï€ when it is estimated from the data, or a given value for Ï€ when it is fixed. The default value is 0.05. When BayesR is used, it is a string seperated by comma where the number of values defines the number of mixture components and each value defines the starting value for each component (the first value is reserved for the zero component); the default values are 0.95,0.03,0.01,0.01. </p>
<p><strong><span class="kwd">--gamma</span></strong> 0,0.01,0.1,1</p>
<p>When BayesR is used, this speficies the gamma values seperated by comma, each representing the scaling factor for the variance of a mixture component. Note that the number of values should match that in <span class="kwd">--pi</span>.</p>
<p><strong><span class="kwd">--hsq</span></strong> 0.5</p>
<p>A starting value for the sampling of SNP-based heritability, which may improve the mixing of MCMC algorithm if it starts with a good estimate. The default value is 0.5.</p>
<p><strong><span class="kwd">--S</span></strong> 0</p>
<p>A starting value for the sampling of the parameter S (relationship between MAF and variance of SNP effects) in BayesS, which may improve the mixing of MCMC algorithm if it starts with a good estimate. The default value is 0.</p>
<p><strong><span class="kwd">--wind</span></strong> 0.2</p>
<p>Specify the window width in Mb for the non-overlapping windows in the nested models, e.g. 0.2 Mb. The default value is 1 Mb.</p>
<p><br></p>
<blockquote>
  <p><strong>Examples</strong></p>
</blockquote>
<p>Standard version of gctb:</p>
<pre><code>gctb --bfile test --pheno test.phen --bayes S --pi 0.1 --hsq 0.5 --chain-length 25000 --burn-in 5000 --out test &gt; test.log 2&gt;&amp;1
</code></pre>
<p>MPI version of gctb (when using intelMPI libraries and two nodes):</p>
<pre><code>mpirun -f $PBS_NODEFILE -np 2 gctb_mpi --bfile test --pheno test.phen --bayes S --pi 0.1 --hsq 0.5 --chain-length 25000 --burn-in 5000 --out test &gt; test.log 2&gt;&amp;1
</code></pre>
<p>The output files include:</p>
<p><strong>test.log</strong>: a text file of running status, intermediate output and final results;</p>
<p><strong>test.snpRes</strong>: a text file of posterior statistics of SNP effects;</p>
<p><strong>test.covRes</strong>: a text file of posterior statistics of covariates;</p>
<p><strong>test.parRes</strong>: a text file of posterior statistics of key model parameters;</p>
<p><strong>test.mcmcsamples.CovEffects</strong>: a text file of MCMC samples for the covariates fitted in the model;</p>
<p><strong>test.mcmcsamples.SnpEffects</strong>: a binary file of MCMC samples for the SNP effects;</p>
<p><strong>test.mcmcsamples.Par</strong>: a text file of MCMC samples for the key model parameters;</p>
<p><br></p>
<blockquote>
  <p><strong>Citations</strong> </p>
</blockquote>
<p><strong>GCTB software and BayesS method</strong><br>
Zeng et al. (2018) Signatures of negative selection in the genetic architecture of human complex traits. 
<em>Nature Genetics</em>, doi: 10.1038/s41588-018-0101-4.</p>
<p><strong>BayesR</strong><br>
Moser et al. (2015) Simultaneous discovery, estimation and prediction analysis of complex traits using a Bayesian mixture model. <a href="http://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1004969"><em>PLoS Genetics</em>, 11: e1004969.</a></p>
<p><strong>BayesN</strong><br>
Zeng et al. (2018) A nested mixture model for genomic prediction using whole-genome SNP genotypes. <a href="http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0194683"><em>PLoS One</em>, 13: e0194683.</a></p>
<p><strong>BayesC`\pi`</strong><br>
Habier et al. (2011) Extension of the Bayesian alphabet for genomic selection. <a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-12-186"><em>BMC Bioinformatics</em>, 12: 186.</a></p>
<p><strong>BayesB</strong><br>
Meuwissen et al. (2001) Prediction of total genetic value using genome-wide dense marker maps. <a href="http://www.genetics.org/content/157/4/1819.short"><em>Genetics</em>, 157: 1819-1829.</a> </p></body></div>
        </div>

        <script>
            $(function() {
                var toc = $("#toc").tocify({
                    selectors: "h2, h3",
                    ignoreSelector: ".notoc",
                    expand: ".expand",
                    showEffect: "fadeIn",
                    scrollHistory: false,
                    extendPage: false,
                    scrollRef: "#PAGE-header",
                    showDIV: "#show",
                    tocID: "#toc"
                }).data("toc-tocify");
                toc.setOptions({
                    appendContents: '<div style="text-align: center;font-size: 14px">Number of visits: <img src="http://hitwebcounter.com/counter/counter.php?page=6809806&style=0011&nbdigits=6&type=page&initCount=0" border="0" width="78"> </div>'
                });
                hljs.initHighlightingOnLoad();

            });

            $(function() {
                setTimeout(function() {
                    (function(i, s, o, g, r, a, m) {
                        i['GoogleAnalyticsObject'] = r;
                        i[r] = i[r] || function() {
                            (i[r].q = i[r].q || []).push(arguments)
                        }, i[r].l = 1 * new Date();
                        a = s.createElement(o),
                            m = s.getElementsByTagName(o)[0];
                        a.async = 1;
                        a.src = g;
                        m.parentNode.insertBefore(a, m)
                    })(window, document, 'script', 'https://www.google-analytics.com/analytics.js', 'ga');
                    ga('create', 'UA-101733990-7', 'auto');
                    ga('send', 'pageview');
                }, 2000);
            });
        </script>
    </div>



</body></html>